### ISSUE TITLE
error compiling immutable dynamic arrays

### ISSUE THREAD

**Issue by tserg** [vyperlang/vyper#2639]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.2+commit.eaef8e7c.dirty
* OS: Linux
* Python Version (output of `python --version`): 3.8.5

### What's your issue about?

Compiling the following contract gives an error
```
# @version ^0.3.2

my_list: immutable(DynArray[uint256, 3])

@external
def __init__():
    my_list = [1, 2, 3]
```
```
vyper.exceptions.ImmutableViolation: Constant value cannot be written to
  contract "contracts/immutableDynArray.vy", function "__init__", line 7:4 
       6 def __init__():
  ---> 7     my_list = [1, 2, 3]
  -----------^
       8
```

### How can it be fixed?

Fill this in if you know how to fix it.

**Comment by charles-cooper** [comment]
will be fixed in https://github.com/vyperlang/vyper/pull/2633?notification_referrer_id=NT_kwDOADsDbbIzMDcxOTA0MTAwOjM4Njc1MDE#issuecomment-1033290113

**Comment by charles-cooper** [comment]
should be fixed in #2633
